db/migrations/99_create_tables_for_batches_of_users.sql

```diff
@@ -0,0 +1,79 @@
+-- +migrate Up
+UPDATE `groups_login_prefixes`
+    SET `prefix` = CONCAT(`prefix`, '_')
+WHERE `prefix` NOT LIKE '%\_';
+
+CREATE TABLE `user_batch_prefixes` (
+    `group_prefix` VARCHAR(13) NOT NULL PRIMARY KEY COMMENT 'Prefix used in front of all batches',
+    `group_id` BIGINT(20) NOT NULL COMMENT 'Group and its subgroups in which managers can create users in batch',
+    `max_users` MEDIUMINT UNSIGNED DEFAULT NULL COMMENT 'Maximum number of users that can be created under this prefix',
+    `allow_new` TINYINT(1) NOT NULL DEFAULT 1 COMMENT 'Whether this prefix can be used for new creations',
+    CONSTRAINT `fk_user_batch_prefixes_group_id_groups_id`
+        FOREIGN KEY (`group_id`) REFERENCES `groups`(`id`) ON DELETE CASCADE
+) ENGINE=InnoDB CHARSET=utf8
+    COMMENT='Authorized login prefixes for user batch creation. A prefix cannot be deleted without deleting batches using it.';
+
+CREATE TABLE `user_batches` (
+    `group_prefix` VARCHAR(13) NOT NULL COMMENT 'Authorized (first) part of the full login prefix',
+    `custom_prefix` VARCHAR(14) NOT NULL
+        COMMENT 'Custom (second) part of the full login prefix',
```

Comment:
-> 'Second part of the full login prefix, given by the user creating the batch'

Reply:
fixed